update-docker-repo-info.py

```diff
@@ -504,11 +504,17 @@ def process_image(image_name, force):
                     os.remove(tag_md_file)
                 REMOVED_IMAGES.append(f"{image_name}:{tag}")
 
+    # if no tags remain for this docker image, remove the entire docker entry
+    if not docker_images_metadata:
```

Comment:
In my opinion its better to move this logic to `save_to_docker_files_metadata_json_file` method instead of check it for each image, just before writing the file, go over the keys under docker_images and remove the empty ones